File: ExpenseTracker.Api/Models/Expense.cs
namespace ExpenseTracker.Api.Models;

public class Expense
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int? CategoryId { get; set; }
    public string Description { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public string CurrencyCode { get; set; } = "USD";
    public DateTime ExpenseDate { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User? User { get; set; }
    public Category? Category { get; set; }
}

File: ExpenseTracker.Web/src/app/core/models/expense.ts
export interface Expense {
  id: number;
  description: string;
  categoryId: number | null;
  category: string;
  amount: number;
  currencyCode: string;
  expenseDate: string;
  notes: string | null;
}

export interface ExpenseRequest {
  description: string;
  categoryId?: number | null;
  amount: number;
  currencyCode: string;
  expenseDate: string;
  notes?: string | null;
}

export interface ExpenseQuery {
  categoryId?: number;
  year?: number;
  month?: number;
}
